### PR TITLE
Add JSON openweathemap api

### PR DESCRIPTION
**Pull request by abrahamcuenca**
* Structured code for the models and controllers
* Setup async calls for downloading network data
* Converted a JSON string to data model
* Utilized the flutter API to download network images
* Displayed a progress indicator while waiting for the time consuming
operations
* Used flutter_dotenv for storing API keys in assets/.env

### CODE REVIEW

**Comment by abrahamcuenca** [comment]
P.s. We have to create a tech debt issue to come back and update tests. 
Going from the mocked UI to UI populated from the API means we have to change how the widget testing is done... but I'm not too sure how to do it yet.

File: lib/ui/Weather.dart [hunk 1]
```diff
@@ -24,7 +29,7 @@ class Weather extends StatelessWidget {
         crossAxisAlignment: CrossAxisAlignment.start,
         children: <Widget>[
           Text(
-            '20',
+            weatherData.temp.toStringAsFixed(0),
```
Comment (Boka44) [inline lib/ui/Weather.dart]: Good idea to set this up early

File: lib/ui/MyHomePage.dart [hunk 2]
```diff
@@ -10,14 +12,41 @@ class MyHomePage extends StatefulWidget {
...
+
   @override
   Widget build(BuildContext context) {
    return new Scaffold(
      backgroundColor: Colors.lightBlue,
       appBar: new AppBar(
        title: Text(widget.title),
       ),
-      body: Weather(),
+      body: _weatherData != null ? Weather(weatherData: _weatherData) :
```
Comment (Boka44) [inline lib/ui/MyHomePage.dart]: Nice loader!

Comment (Boka44) [reply lib/ui/MyHomePage.dart]: I would suggest storing ternary operators in separate variables for cleaner code.

File: lib/ui/Weather.dart (hunk 1 above)
Comment (abrahamcuenca) [reply lib/ui/Weather.dart]: Thanks!

File: lib/main.dart [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
+  runApp(MyApp());
+}
 
 class MyApp extends StatelessWidget {
   // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
-      title: 'Flutter Demo',
+      title: 'Weather App',
```
Comment (Boka44) [inline lib/main.dart]: It's important to name your app!

File: lib/ui/MyHomePage.dart (hunk 2 above)
Comment (abrahamcuenca) [reply lib/ui/MyHomePage.dart]: Thanks! I didnt mean to resolve conversation haha I was typing a response and hitting enter which was 'resolving' the conversation... pull request for github code review? haha jk

Comment (abrahamcuenca) [reply lib/ui/MyHomePage.dart]: Can you provide an example of storing the ternary operations in separate variables? like a quick code snippet?

File: lib/main.dart (hunk 3 above)
Comment (abrahamcuenca) [reply lib/main.dart]: Yeah hahaha!

**Pull request merged**